### ISSUE TITLE
Parallel Convergence failure with 9 MPI ranks, but success with fewer ranks

### ISSUE THREAD

**Issue by AndreasRIHU** [MODFLOW-USGS/modflow6#1317]
Hi,
the attached model (one of your test models) shows an unexplicable behaviour in  parallel mode:

[__test_convergence_err.zip](https://github.com/MODFLOW-USGS/modflow6/files/12217616/__test_convergence_err.zip)

when calling it inside a WSL2 bash like that
  mpiexec -np 9 --use-hwthread-cpus "/home/arost/modflow6/bin/mf6" --PARALLEL
it produces a convergence failure.
But when calling it with np=8 (6,3,2 tested too), it converges well.
I think, that this is a bug, because it has got 9 submodels and thus should run with 9 ranks as well.

Best regards
Andreas

**Comment by jdhughes-usgs** [comment]
The problem converges for me on MacOS in 13.7 seconds using

`mpiexec -n 9 ./mf6 -p` 

One limitation of the current version is IMS settings for the linear solver are not transferred to petsc. So we include a .petscrc file in the directory with the `mfsim.nam` file to set the linear solver settings for petsc. With the .petscrc file the problem converges for me in 1.6 sec.

The contents of the .petscrc file for this problem is

```
-ksp_type cg
-pc_type bjacobi
-sub_pc_type ilu
-sub_pc_factor_levels 2
-dvclose 1.0e-6
-options_left no
```

For a model using Newton-Raphson your .petscrc file would be

```
-ksp_type bcgs
-pc_type bjacobi
-sub_pc_type ilu
-sub_pc_factor_levels 2
-dvclose 1.0e-6
-options_left no
```

We are hoping to transfer IMS settings common to petsc prior to the next release.

**Comment by langevin-usgs** [comment]
Have you tried specifying the --oversubscribe argument for mpiexec?  This might get it to run, though it may not perform well.

**Comment by AndreasRIHU** [comment]
Hi Chris,
I've just tried --oversubscribe both with and without --use-hwthread-cpus.
But it still shows that behaviour:
  np9: no convergence
  np8 and below: convergence as expected...
Best regards
Andreas

**Comment by AndreasRIHU** [comment]
I've just tried the second suggestion from [jdhughes-usgs](https://github.com/jdhughes-usgs) (-ksp_type bcgs) even though the model doesn't use Newton Raphson formulation.
With that, even 9 MPI ranks converge successfully.
So for me it looks like the cgs path in Petsc is creating the problem?
Regards
Andreas

**Comment by mjr-deltares** [comment]
Hi @AndreasRIHU. OK, I will look into this, the problem being that the IMS solver converges for CG but the PETSc doesn't. I just checked, and even on 1 process the PETSc solver fails to converge. Will let you know when there's an update. For now I guess you could use the `-ksp_type bcgs` setting to continue with your work.

**Comment by AndreasRIHU** [comment]
Hi mjr-deltares,
 
sounds good.
I've postponed testing the parallel feature until the next version is released, due to some other issues, its alpha status and until more documentation will be available.
This should be the case in late autumn - according to Chris.
 
Thanks
Andreas

**Comment by mjr-deltares** [comment]
Hi @AndreasRIHU,
I see, thanks for testing so early in the development though! It's a delicate line, you know. We do value your feedback at this stage very much but the resolution of reported issues might still be relatively far ahead...
Best,
Martijn

**Comment by mjr-deltares** [comment]
CG with PETSc is on the development list, closing this issue

**Comment by AndreasRIHU** [comment]
Hi Martijn,
 
after some months have passed, I've been testing the latest Modflow6 development version in parallel again.
It already works better - congrats!
Especially the prebundled Windows package at your nightly builds makes it much easier to use.
I've got some crashes / issues but wanted to ask beforehand, if it already makes sense again to report those or if it's still to early for you?
 
Best regards
Andreas

Thursday, September 7, 2023, 3:00:20 PM, you wrote:
